### ISSUE TITLE
Proxmox-iso builder tries to use IPv6 address when not applicable

### ISSUE THREAD

**Issue by mkbrown** [hashicorp/packer#10227]
Packer v 1.6.5 Linux,  Proxmox-iso builder

Windows Server 2019 Guest on Proxmox 6.2-12
Qemu-Guest Agent v100.0.0

Host running Packer has only an IPv4 address, and packer http server bound to 0.0.0.0.  Packer insists on trying to connect with the VM using the first detected IP, which is the IPv6 address.  So, the packer never connects to run the provisioners, and the CI job I'm running the build under times out.

Please have the packer proxmox-iso builder prefer an IP address in the same range as that on the host running packer.

```
2020/11/06 16:37:48 packer-builder-proxmox plugin: [DEBUG] Error getting WinRM host: error unmarshalling result Could not parse fe80::b9a0:e82:8476:a3fe%5 as IP
```
![image](https://user-images.githubusercontent.com/10076016/98417402-d8b1c900-204e-11eb-9567-36967003e2d4.png)

Thanks!

**Comment by nywilken** [comment]
Hi there @mkbrown pinging @carlpett for some assistance. By default packer will select the first non-loopback interface. Some builders have options to help Packer with the interface selection. 

Have you tried using `vm_interface` or `http_interface` to let Packer know that it should select the IPV4 interface? 

Here's a link to the docs for the two options above https://www.packer.io/docs/builders/proxmox/iso#vm_interface

**Comment by mkbrown** [comment]
@nywilken I haven't set either vm_interface or http_interface, as the docs state:

http_interface - (string) - Name of the network interface that Packer gets HTTPIP from. Defaults to the first non loopback interface.
vm_interface - (string) - Name of the network interface that Packer gets the VMs IP from. Defaults to the first non loopback interface.

I have packer running via gitlab-runner in an LXC container on the Proxmox host.  It is grabbing the only non-loopback interface, but is trying the IPv6 address, and not trying the IPv4 address.  

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0@if106: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 00:16:3e:38:06:80 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.30.30.237/24 brd 172.30.30.255 scope global noprefixroute eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::216:3eff:fe38:680/64 scope link 
       valid_lft forever preferred_lft forever

Is it possible to configure the vm_interface setting to use the IPv4 address rather than the interface name?

Thanks for your assistance!

**Comment by paginabianca** [comment]
Hi, sorry that I'm a bit late to this issue.
The Proxmox builder does not distinguish between the IPv6/v4 address of an interface. It just takes the first one that gets returned by the Promxmox API. So it's more or less random?(I'm not sure how things work on the proxmox side).

If I find the time, I'll have a look at it this weekend. It shouldn't take too much effort.
If you want to have a loot at the code yourself, the part you're looking for is [this](https://github.com/hashicorp/packer/blob/master/builder/proxmox/common/builder.go) at line 130.

**Comment by mmassez** [comment]
The problem is 2-fold.
The used API client for Proxmox (Telmate/proxmox-api-go) cannot parse IPv6 addresses with a zone id (delimited by "%").
I've wrote a simple fix which splits the IPv6 address at the delimiter "%" and returns the first part.
That fix can be checked here: https://github.com/Telmate/proxmox-api-go/pull/112

Secondly I've added an extra check in the proxmox builder.go which basically tries to parse all IP addresses to IPv4 and thus returns the first valid IPv4 address.

I'll make the pull request.
